src/pepr/operator/crd/validators/package-validator.ts

```diff
@@ -92,6 +92,27 @@ export async function validator(req: PeprValidateRequest<UDSPackage>) {
         `The client ID "${client.clientId}" uses an invalid secret name ${client.secretName}`,
       );
     }
+    // If standardFlowEnabled is undefined (defaults to `true`) or explicitly true and there are no redirectUris set, deny the req
+    if (client.standardFlowEnabled !== false && !client.redirectUris) {
+      return req.Deny(
+        `The client ID "${client.clientId}" must specify redirectUris if standardFlowEnabled is turned on`,
+      );
+    }
+    // If this is a public client ensure that it only sets itself up as an OAuth Device Flow client
+    if (
+      client.publicClient &&
+      (client.standardFlowEnabled !== false ||
+        client.secret !== undefined ||
+        client.secretName !== undefined ||
+        client.secretTemplate !== undefined ||
+        client.enableAuthserviceSelector !== undefined ||
+        client.protocol === "saml" ||
+        client.attributes?.["oauth2.device.authorization.grant.enabled"] !== "true")
+    ) {
+      return req.Deny(
+        `The client ID "${client.clientId}" must _only_ configure the OAuth Device Flow as a public client`,
```

Comment:
This denial message is relatively clear but it might be useful to document the intended usage of Device Flow (and the corresponding toggles for `publicClient` and `standardFlowEnabled`). Hopefully at least that way it is documented what the intent for these toggles are and people are less likely to get through most a deploy before it bombs out.

Reply:
Why do _I_ have to write all the docs and tests :sob: :sweat_smile:

Reply:
Please see checklist on PR description: ` Test, docs, adr added or updated as needed`